RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -281,13 +260,40 @@ public Optional<ObjectNode> updateDocument(String indexName, String documentId,
             .retryWhen(Retry.backoff(3, Duration.ofSeconds(1)).maxBackoff(Duration.ofSeconds(10)))
             .block();
         if (response.code == HttpURLConnection.HTTP_OK) {
-            return Optional.of(body);
+            return Optional.of(newBody);
```

Comment:
This should be from the response body since pipelines or other systems could alter the body

Reply:
I agree - but in my testing, it wasn't in the response body, so I did things this way instead.

See: https://opensearch.org/docs/2.11/api-reference/document-apis/update-document/#response

Reply:
If the function returns the old version is seq number will be wrong, no?  If you attempted to perform two updates on the same the second one would fail even if you used the output of this function.